### PR TITLE
ammonite: Add installCheckPhase

### PR DESCRIPTION
**Pull request by NeQuissimus**
###### Motivation for this change
Add testing

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NeQuissimus** [comment]
@ofborg test ammonite

File: pkgs/development/tools/ammonite/default.nix [hunk 1]
```diff
@@ -17,7 +17,7 @@ let
         inherit sha256;
       };
 
-      phases = "installPhase";
+      phases = [ "installPhase" "installCheckPhase" ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/ammonite/default.nix:20]: Please never overwrite phases and disable the one you do not want to use.

File: pkgs/development/tools/ammonite/default.nix [hunk 2]
```diff
@@ -1,5 +1,5 @@
-{ stdenv, fetchurl, jre, nixosTests, writeScript, common-updater-scripts, git
-, nixfmt, nix, coreutils, gnused, disableRemoteLogging ? true }:
+{ stdenv, lib, fetchurl, jre, nixosTests, writeScript, common-updater-scripts, git
+, nixfmt, nix, coreutils, gnused, util-linux, disableRemoteLogging ? true }:
 
 with stdenv.lib;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/ammonite/default.nix:4]:
```suggestion
with lib;
```

Also I would prefer if lib is moved to the places it is needed.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110517` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>3 packages failed to build and are new build failures:</summary>
  <ul>
<li>ammonite: log <a href="https://termbin.com/wus1">https://termbin.com/wus1</a></li>
<li>(ammonite-repl): log was empty</li>
<li>ammonite_2_12: log <a href="https://termbin.com/b7ns">https://termbin.com/b7ns</a></li>
  </ul>
</details>

**Comment by rmcgibbo** [comment]
I think it's possible that there's an infinite loop here, but I'm not sure if that's really true or if so, why. I've had this build running for about 2 hours at 100% CPU usage, and it's still going.

**Comment by NeQuissimus** [comment]
@ofborg test ammonite

**Comment by NeQuissimus** [comment]
@rmcgibbo That's why this was still a draft PR. Something was whacky but I hadn't figured it out :D
`tput` was missing, so the output was infinite :)

**Comment by SuperSandro2000** [comment]
@NeQuissimus build failed.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110517` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>3 packages failed to build and are new build failures:</summary>
  <ul>
<li>ammonite: log <a href="https://termbin.com/jdsp">https://termbin.com/jdsp</a></li>
<li>(ammonite_2_13): log was empty</li>
<li>ammonite_2_12: log <a href="https://termbin.com/oilr">https://termbin.com/oilr</a></li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110517` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>3 packages failed to build and are new build failures:</summary>
  <ul>
<li>ammonite: log <a href="https://termbin.com/nlbg">https://termbin.com/nlbg</a></li>
<li>(ammonite_2_13): log was empty</li>
<li>ammonite_2_12: log <a href="https://termbin.com/8rtb">https://termbin.com/8rtb</a></li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110517` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>ammonite (ammonite_2_13)</li>
<li>ammonite_2_12</li>
</ul>
</details>

**Comment by NeQuissimus** [comment]
OK, something is up with this on Darwin. I will take a look

**Pull request closed without merging**